### PR TITLE
Get logs from pvc helper pods when job fails

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
Add `#getPodLogs()` method to KubernetesDeployments. This method is used to get logs of failed PVC subpath helper jobs and log them to aid in debugging PVC issues.

For example, when #12445 occurs, the message 
`2019-01-28 18:53:23,610[er-ThreadPool-0]  [ERROR] [e.c.w.i.k.n.p.PVCSubPathHelper 150]  - Job command '[rm, -rf, /tmp/job_mount/workspacegij0p3l3wwio2g3j]' execution is failed. Logs: rm: cannot remove '/tmp/job_mount/workspacegij0p3l3wwio2g3j/projects': Permission denied \nrm: cannot remove '/tmp/job_mount/workspacegij0p3l3wwio2g3j/m2': Permission denied \nrm: cannot remove '/tmp/job_mount/workspacegij0p3l3wwio2g3j/javadata': Permission denied \nrm: cannot remove '/tmp/job_mount/workspacegij0p3l3wwio2g3j/che-logs/dev-machine': Permission denied \n`
is logged.

### What issues does this PR fix or reference?
Related to https://github.com/eclipse/che/issues/12445

### CODE REVIEW

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java [hunk 1]
```diff
@@ -653,6 +653,29 @@ public void exec(String name, String containerName, int timeoutMin, String[] com
     }
   }
 
+  /**
+   * Get logs from pod with specified name. The name can refer to either a deployment or pod
+   * directly. In case of a deployment being provided, logs are returned from pods in that
+   * deployment.
+   *
+   * @returns the logs from the pod or null if pod cannot be found.
```
Comment (sleshchenko) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java]: Should not `@returns` be `@return`?

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/PVCSubPathHelper.java [hunk 2]
```diff
@@ -145,10 +146,11 @@ void execute(String workspaceId, String[] commandBase, String... arguments) {
       final Pod finished = deployments.wait(podName, WAIT_POD_TIMEOUT_MIN, POD_PREDICATE::apply);
       PodStatus finishedStatus = finished.getStatus();
       if (POD_PHASE_FAILED.equals(finishedStatus.getPhase())) {
+        String logs = deployments.getPodLogs(podName);
```
Comment (sleshchenko) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/PVCSubPathHelper.java:149]: Please consider adding the corresponding `verify` line in `PVCSubPathHelperTest`

**Review by sleshchenko — APPROVED** [review]
LGTM

It would be nice to see in PR description example of logged message

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java [hunk 3]
```diff
@@ -653,6 +653,29 @@ public void exec(String name, String containerName, int timeoutMin, String[] com
...
+   */
+  public String getPodLogs(String name) {
```
Comment (sleshchenko) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java:665]: Please consider adding `@Nullable` annotation here

**Review by ibuziuk — APPROVED** [review]

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java (hunk 1 above)
Comment (amisevsk) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java]: Good catch, thanks. Fixed.

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/PVCSubPathHelper.java (hunk 2 above)
Comment (amisevsk) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/PVCSubPathHelper.java:149]: Done.

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java (hunk 3 above)
Comment (amisevsk) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java:665]: Done.

**Review by garagatyi — APPROVED** [review]

**Comment by amisevsk** [comment]
@dneary This PR doesn't address PVC issues directly -- it just adds some logging for when our PVC jobs fail. It's helpful for debugging on our side but would generally be invisible to a Che user. 

Some of the issues @sleshchenko is working on may add this, but sadly this one is just for our convenience.

OTOH, I did add a commit to [this PR](https://github.com/eclipse/che/pull/12486#issuecomment-458304268) that adds/improves recipe validation, which could be helpful to an end-user.

**Pull request merged**